docs/vm-configuration.md

```diff
@@ -0,0 +1,265 @@
+they allow to create virtual GPU devices, or NPIV virtual SCSI host adapters
+which can then be assigned to the guest.
+
+
+Design implications
+-------------------
+
+The key take away from the analysis above is that kubevirt needs to maintain a
+strict separation between the libvirt XML document format, and the format used
+in the public APIs. The tenant users who are specifying the virtual machine
+configuration cannot be permitted to have direct control over most aspects of
+backend configuration. This is because they are both untrusted users, and do not
+have knowledge about the compute hosts which is required in order to make the
+correct choices.
+
+The public k8s virtual machine configuration will must allow users to fully
+specify the frontend virtual hardware, in order to enable a stable ABI to be
+attained.
+
+For the backend configuration, some parts be can automatically determined by
+kubevirt without requiring any user specification. For example, there is no need
+for the user to specify what VNC/SPICE port to use, or where to find the UEFI
+BIOS files. It is sufficient to know that VNC or UEFI must be enabled. Other
+areas of backend configuration will require the user to specify resources in an
+abstracted manner. For example, instead of providing a full file path for a disk,
+the tenant would provide the name of a PersistentVolumeClaim resource, or
+alternative k8s API resource. Further areas of backend configuration will
+require users to specify desired goals, which kubevirt then translates into
+specific configuration information. For example, the tenant may request exclusive
+use of CPUs, and kubevirt has to decide which host CPUs to bind virtual CPUs to,
+potentially impacting NUMA node placement of the guest as a result.
+
+Given a k8s API resource describing the virtual machine, it will be possible to
+generate a libvirt domain XML document from scratch. The particular API usage
+scenario, however, may affect what information is provided in the domain XML.
+The domain XML generated at the time a guest is started, will be the most fully
+specified XML in terms of both frontend and backend configuration. If domain XML
+is generated in isolation, however, while the frontend can be fully specified,
+the backend configuration may remain partially specified. For example, it would
+be possible to expand a PVC to a backend disk config, but without knowledge of
+a specific host to run the guest, the domain XML cannot contain information
+about CPU pinning.
+
+Going in the reverse direction is more difficult in general. The conversion
+from the high level k8s resource to the domain XML format is a lossy process.
+So given a domain XML document in isolation, it is not possible to generate a
+full k8s resource description of the virtual machine. At most it can contain
+frontend information. Translating the backend information in the domain XML, to
+the backend information required in the k8s resource is impractical or impossible
```

Comment:
The source VM still needs to have a full representation in kubernetes though. ie we need a VM spec for the currently running source VM, referencing all the required k8s resources for disk/network/etc. The virt-handler needs to be fully tracking this source VM, in the normal manner. You can't *live* migrate a source VM whose definition lives totally outside k8s, into a new VM that lives inside k8s, which is what I read @mykaul 's comment as suggesting. 

Now there's nothing stopping your *offline* migrating a VM into k8s - that's just a matter of something creating suitable manifests to represent the config that oVirt would have used, so when kubevirt launches it, it matches the original frontend state.  This is outside scope of kubevirt though - something for oVirt to tackle.